### PR TITLE
Add DataTransfer article

### PR DESCRIPTION
**Pull request by tomayac**
Changes proposed in this pull request:

- Add DataTransfer article.

### CODE REVIEW

**Comment by tomayac** [comment]
CC: @jtangelder.

File: src/site/content/en/blog/datatransfer/index.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+included in the transferred data.
+
+Think of all the possibilities when you can transfer data between any (web) application with no
+restrictions. Sharing events from a calendar to your favorite ToDo app, attaching virtual files to
+emails, sharing contacts. That would be great, right? This starts with you! 🙌
+
+## Concerns
+
+While transferring data between web applications with DataTransfer is available and ready to use
+today, there are some things to be aware of before integrating.
```
Comment (jpmedley) [inline src/site/content/en/blog/datatransfer/index.md:199]:
```suggestion
While DataTransfer is available today, there are some things to be aware of before integrating.
```

File: src/site/content/en/blog/datatransfer/index.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+### Browser compatibility
+
+Desktop browsers all have great support for the technique described above, while mobile devices have
```
Comment (jpmedley) [inline src/site/content/en/blog/datatransfer/index.md:203]:
```suggestion
Desktop browsers all have great support for the technique described above, while mobile devices do
```

File: src/site/content/en/blog/datatransfer/index.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+to desktop browsers only.
+
+### Discoverability
+
+Drag-drop and copy-paste are system-level interactions when working on a desktop computer, with
+roots back to the first GUIs about 40 years ago. Think for example about how many times you have
+used these interactions for organizing files. On the web, this is not very common yet.
+
+You will need to educate users about this new interaction, and come up with UX patterns to make this
+recognizable.
```
Comment (jpmedley) [inline src/site/content/en/blog/datatransfer/index.md:216]: I've heard that there are even young people who's experience with computers is confined to mobile devices. Do you need to add some language starting with, "That many mean more than telling them it's like desktop."

File: src/site/content/en/blog/datatransfer/index.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+- Web applications you are dragging over have access to the type keys, not the data. The data only
+  becomes available on drop or paste.
+- The received data should be treated like any other user input; sanitize and validate before using.
+
+## Getting started with DataTransfer!
+
+Are you excited about using the DataTransfer in your application? Consider taking a look at the
+[Transmat library on GitHub](google.github.io/transmat). This open-source library aligns browser
+differences, provides JSON-LD utilities, contains an observer to respond to transfer events for
+highlighting drop-areas, and lets you integrate DataTransfer along existing drag and drop
```
Comment (jpmedley) [inline src/site/content/en/blog/datatransfer/index.md:238]:
```suggestion
highlighting drop-areas, and lets you integrate DataTransfer among existing drag and drop
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: src/site/content/en/blog/datatransfer/index.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+  - progressive-web-apps
+---
+
+You might have heard about the
+[DataTransfer API](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer) before that is
+part of the
+[HTML5 Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API)
+and [Clipboard events](https://developer.mozilla.org/en-US/docs/Web/API/Element/copy_event). It can
+be used to transfer data between source and receiving targets. This API is available and
+[ready to use](https://caniuse.com/mdn-api_datatransfer_setdata) in all modern desktop browsers.
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:24]: Completely redundant. Removed :-)

File: src/site/content/en/blog/datatransfer/index.md [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+
+The drag-drop and copy-paste interactions are often used for interactions within the page,
+transferring a simple text from A to B. But what is oftentimes overlooked is the ability to use
+these same interactions to go beyond the browser window.
+
+Both the browser's built-in drag-and-drop as well as the copy-paste interactions can communicate
+with other (web) applications, not tied to an origin. The API has support for providing multiple
```
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:31]:
```suggestion
with other (web) applications, not tied to any origin. The API has support for providing multiple
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:31]: Good catch. "Not tied to _any_ origin" actually.

File: src/site/content/en/blog/datatransfer/index.md [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+data entries that can have different behavior based on where it is transferred to. Your
+web application can send and receive the transferred data when listening to incoming events.
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:33]: Definitely better. Thanks!

File: src/site/content/en/blog/datatransfer/index.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+</figure>
+
+## Transferring data
+
+To get started with transferring data, you'll need to implement drag-drop or copy-paste. The examples
+below will be based on drag-drop interactions, but the process for copy-paste is very similar. If
+you are unfamiliar with the Drag and Drop API, there's a great article
+[explaining HTML5 Drag and Drop](https://web.dev/drag-and-drop/) that details the ins and outs.
+
+By providing MIME-type keyed data, you are able to interact with external applications, for free!
```
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:52]:
```suggestion
By providing MIME-type keyed data, you are able to interact with external applications.
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:52]: Hahah, free not as in beer, but as in freedom. Rephrased.

File: src/site/content/en/blog/datatransfer/index.md [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+  contentUrl: 'venice.jpg',
+  datePublished: '2010-08-08',
+  description: 'I took this picture during our honey moon.',
+  name: 'Canal in Venice',
+};
+event.dataTransfer.setData('application/ld+json', JSON.stringify(data));
+```
+
+When using the Schema.org types, you can start with the generic [Thing](schema.org/Thing) type, but
+go down to types like [Event](schema.org/Event), [Person](schema.org/Person),
```
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:182]:
```suggestion
When using the Schema.org types, you can start with the generic [Thing](schema.org/Thing) type,
or use something closer to your use case like [Event](schema.org/Event), [Person](schema.org/Person),
```

File: src/site/content/en/blog/datatransfer/index.md [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+[MediaObject](schema.org/MediaObject), [Place](schema.org/Place), and even
+[MedicalEntity](schema.org/MedicalEntity) if you like. If you use TypeScript, you can use the
```
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:184]:
```suggestion
[MediaObject](schema.org/MediaObject), [Place](schema.org/Place), or even highly-specific types like
[MedicalEntity](schema.org/MedicalEntity) if need be. When you use TypeScript, you can use the
```

File: src/site/content/en/blog/datatransfer/index.md (hunk 3 above)
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:216]:
```suggestion
recognizable, especially for people who's experience with computers so far has been confined to mobile devices.
```

Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:216]: Added as a by-phrase.

File: src/site/content/en/blog/datatransfer/index.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+The drag-drop and copy-paste interactions are often used for interactions within a page,
+transferring a simple text from A to B. But what is oftentimes overlooked is the ability to use
+these same interactions to go beyond the browser window.
+
+Both the browser's built-in drag-and-drop as well as the copy-paste interactions can communicate
+with other (web) applications, not tied to any origin. The API has support for providing multiple
+data entries that can have different behaviors based on where data is transferred to. Your
+web application can send and receive the transferred data when listening to incoming events.
+
+This powerful capability can change the way we think about sharing and interoperability in web
```
Comment (tomayac) [inline src/site/content/en/blog/datatransfer/index.md:35]:
```suggestion
This capability can change the way we think about sharing and interoperability in web
```

File: src/site/content/en/blog/datatransfer/index.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+The drag-drop and copy-paste interactions are often used for interactions within the page,
+transferring a simple text from A to B. But what is oftentimes overlooked is the ability to use
+these same interactions to go beyond the browser window.
+
+Both the browser's built-in drag-and-drop as well as the copy-paste interactions can communicate
+with other (web) applications, not tied to an origin. The API has support for providing multiple
+data entries that can have different behavior based on where it is transferred to. Your
+web application can send and receive the transferred data when listening to incoming events.
+
+This powerful capability can change the way we think about sharing and interoperability in web
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/datatransfer/index.md:35]: Let's just remove the word then.

**Comment by jtangelder** [comment]
LGTM!

**Pull request merged**